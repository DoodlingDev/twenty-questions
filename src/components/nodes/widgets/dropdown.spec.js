/* eslint require-jsdoc: "off" */
import React from "react";
import Dropdown from "./dropdown";
import { mount } from /* mount */ "enzyme";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const defaultProps = {
  label: "test label",
  path: "this.test.dropdown",
  name: "test_dropdown",
  valueManager: {
    values: {},
    validate: {},
    update: jest.fn(),
  },
  options: [
    "hello",
    "world"
  ],
};

describe("rendering", () => {
  it("renders correctly", () => {
    const render = renderer.create(<Dropdown {...defaultProps}/>).toJSON();
    expect(render).toMatchSnapshot();
  });

  it("renders correctly with no options supplied", () => {
    const noOptionsProps = {
      label: "test label",
      path: "this.test.dropdown",
      name: "test_dropdown",
      valueManager: {
        values: {},
        validate: {},
      },
    };
    const render = renderer.create(<Dropdown {...noOptionsProps}/>).toJSON();
    expect(render).toMatchSnapshot();
  });

  it("renders correctly with no label supplied", () => {
    const noLabelProps = {
      path: "this.test.dropdown",
      name: "test_dropdown",
      valueManager: {
        values: {},
        validate: {},
      },
      options: [
        "hello",
        "world"
      ],
    };
    const render = renderer.create(<Dropdown {...noLabelProps}/>).toJSON();
    expect(render).toMatchSnapshot();
  });
});

function setup(renderFn, testProps = defaultProps) {
  return renderFn(<Dropdown {...testProps}/>);
}

describe("interactions", () => {
  it("should use valueManager to update on change", () => {
    const wrapper = setup(mount);
    const event = {
      target: {
        value: "selected text",
      },
    };
    wrapper.find("select").first().simulate("change", event);
    expect(wrapper.props().valueManager.update.mock.calls[0]).toMatchObject(
      [ { path: "this.test.dropdown",
          name: "test_dropdown",
          value: "selected text" } ]
    );
  });
});



