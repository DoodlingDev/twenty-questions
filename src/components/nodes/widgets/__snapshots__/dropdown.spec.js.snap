// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="q20_node__string--dropdownWidget"
>
  <div
    className="q20_node__testDropdown--labelAndDescription"
  >
    <label
      className="q20_node__testDropdown--label"
      htmlFor="this.test.dropdown"
    >
      test label
    </label>
  </div>
  <div
    className="q20_testDropdown__errorHandler"
  >
    <select
      className="q20_node__thisTestDropdown--dropdownField"
      onChange={[Function]}
      type="select"
      value={undefined}
    >
      <option
        value=""
      >
        test label
      </option>
      <option
        value="hello"
      >
        hello
      </option>
      <option
        value="world"
      >
        world
      </option>
    </select>
  </div>
</div>
`;

exports[`rendering renders correctly with no label supplied 1`] = `
<div
  className="q20_node__string--dropdownWidget"
>
  <div
    className="q20_node__testDropdown--labelAndDescription"
  />
  <div
    className="q20_testDropdown__errorHandler"
  >
    <select
      className="q20_node__thisTestDropdown--dropdownField"
      onChange={[Function]}
      type="select"
      value={undefined}
    >
      <option
        value="hello"
      >
        hello
      </option>
      <option
        value="world"
      >
        world
      </option>
    </select>
  </div>
</div>
`;

exports[`rendering renders correctly with no options supplied 1`] = `
<div
  className="q20_node__string--dropdownWidget"
>
  <div
    className="q20_node__testDropdown--labelAndDescription"
  >
    <label
      className="q20_node__testDropdown--label"
      htmlFor="this.test.dropdown"
    >
      test label
    </label>
  </div>
  <div
    className="q20_testDropdown__errorHandler"
  >
    <select
      className="q20_node__thisTestDropdown--dropdownField"
      onChange={[Function]}
      type="select"
      value={undefined}
    >
      <option
        value=""
      >
        test label
      </option>
    </select>
  </div>
</div>
`;
